pkg/controllers/localbuild/nginx.go

```diff
@@ -0,0 +1,134 @@
+package localbuild
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"time"
+
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+	"github.com/cnoe-io/idpbuilder/pkg/k8s"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	nginxNamespace  string = "ingress-nginx"
+	nginxDeployment string = "ingress-nginx-controller"
+)
+
+//go:embed resources/nginx/k8s/*
+var installNginxFS embed.FS
+var timeout = time.After(5 * time.Minute)
```

Comment:
5 minutes seems a bit too much? Users may end up needing to wait 5 minutes just to see something went wrong. 
Maybe one minute?

Reply:
it could take more than one minute for the ingress controller to become ready. i'll bring it down though to maybe 2 or 3 mins and see how that works

Reply:
Sounds good.